### PR TITLE
Add TypeScript declarations to module

### PR DESCRIPTION
**Pull request by hedss**
Connects to #22.

### CODE REVIEW

**Comment by hedss** [comment]
@pimterry @emirotin @Page- @lekkas 
First attempt at adding TypeScript declarations (I'm trying to do this on a small, innocuous module first, given I need to do this for several other bigger modules in `rdt` for new functionality being added).

Guides followed were https://www.typescriptlang.org/docs/handbook/declaration-files/templates/module-class-d-ts.html & https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html

Let me know what I'm doing wrong/right!

File: lib/declarations.d.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+declare module "resin-discoverable-services" {
+    export interface publishOptions {
```
Comment (pimterry) [inline lib/declarations.d.ts]: Interface names would normally be capitalized.

File: lib/declarations.d.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+        identifier: string;
+        name: string;
+        host?: string;
+        port: number;
+    }
+
+    export function setRegistryPath(path: string): void;
+    export function enumerateServices(callback?: (error: Error, results: [string]) => void): Promise<[string]>;
+    export function findServices(services: [string], timeout?: number, callback?: (error: Error, results: [string]) => void): Promise<[string]>;
+    export function publishServices(services: [publishOptions], callback?: (error: Error) => void): Promise<void>;
```
Comment (pimterry) [inline lib/declarations.d.ts]: You want`string[]`, not `[string]`.

`string[]` literally says "this is an array of strings".

`[string]` says "this is a tuple containing one string element". [Tuple syntax](https://www.typescriptlang.org/docs/handbook/basic-types.html#tuple) lets you write types like `[number, string, string]`. Here you've used just one element, and TypeScript then lets you go outside the tuple indexes later and uses the union of the tuple types (which here is just `string`). That all means that, yes, this does technically act the same as an array of strings, but it's a pretty weird way of describing that.

File: lib/declarations.d.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+declare module "resin-discoverable-services" {
+    export interface publishOptions {
+        identifier: string;
+        name: string;
+        host?: string;
+        port: number;
+    }
+
+    export function setRegistryPath(path: string): void;
+    export function enumerateServices(callback?: (error: Error, results: [string]) => void): Promise<[string]>;
```
Comment (emirotin) [inline lib/declarations.d.ts]: `error` should be nullable as well as the result, right?

And looks like Tim's comment about arrays applies everywhere here

Comment (emirotin) [inline lib/declarations.d.ts]: Also I would create a generic node-style callback type, like `type Callback < T > = (error?: Error, result?: T[]) => void;`

File: lib/declarations.d.ts (hunk 2 above)
Comment (hedss) [inline lib/declarations.d.ts]: Thanks Tim, I knew it didn't look right.

**Pull request merged**